src/main/src/test/java/org/geoserver/config/GeoServerPropertyConfigurerTest.java

```diff
@@ -35,6 +37,17 @@ public void tearDown() throws Exception {
         ctx.close();
     }
 
+    @Test
+    public void testConfigFileCreated() throws IOException {
```

Comment:
Is this testing anything relevant to the rest of the PR?

Reply:
Did we decided if this is needed here?

Reply:
Took a while to realize it again. If you roll back to b14f54e1 testConfigFileCreated you can see the test to snap:

Without throwing a `FileNotFoundException` [GeoServerPropertyConfigurer.loadProperties()](https://github.com/geoserver/geoserver/blob/main/src/main/src/main/java/org/geoserver/config/GeoServerPropertyConfigurer.java#L104) won't ever enter the catch block, since an empty file was silently created, which did not hurt anybody else before.

The (otherwise failed) creation takes place within [GeoServerPropertyConfigurerTest.setUp()](https://github.com/geoserver/geoserver/pull/6314/files#diff-6e0586e9277979b3436a1ee5aa890935b97522db04e285e655279ce2d4ee236fR31) outside of testConfigFileCreated() and is hard to locate, since the exception is caught. Here is a stacktrace of the hidden exeption:

<pre>
java.io.FileNotFoundException: foo.properties
	at org.geoserver.config.SpringResourceAdaptor.getInputStream(SpringResourceAdaptor.java:62)
	at org.springframework.core.io.support.EncodedResource.getInputStream(EncodedResource.java:159)
	at org.springframework.core.io.support.PropertiesLoaderUtils.fillProperties(PropertiesLoaderUtils.java:99)
	at org.springframework.core.io.support.PropertiesLoaderSupport.loadProperties(PropertiesLoaderSupport.java:182)
--->	at org.geoserver.config.GeoServerPropertyConfigurer.loadProperties(GeoServerPropertyConfigurer.java:103)
	at org.springframework.core.io.support.PropertiesLoaderSupport.mergeProperties(PropertiesLoaderSupport.java:163)
	at org.springframework.context.support.PropertySourcesPlaceholderConfigurer.postProcessBeanFactory(PropertySourcesPlaceholderConfigurer.java:144)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:291)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:167)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:707)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:533)
	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:202)
	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:177)
	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:163)
	at org.geoserver.config.GeoServerPropertyConfigurerTest.setUp(GeoServerPropertyConfigurerTest.java:31)
....
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
</pre>

So, after introducing the FNF Exception, creation of the missing property file works now and the unit test verifies it. I think this was hidden bug and went wrong before.